daemon/api.go

```diff
@@ -577,27 +578,53 @@ func waitChange(chg *state.Change) error {
 	return chg.Err()
 }
 
+func ensureUbuntuCore(st *state.State, chg *state.Change, other *state.TaskSet) error {
+	var ss snapstate.SnapState
+	var lastCoreTask *state.Task
+
+	ubuntuCore := "ubuntu-core"
+	err := snapstateGet(st, ubuntuCore, &ss)
+	if err != state.ErrNoState {
+		return nil
```

Comment:
I think we want `return err` here (if it's nil, great, if not, it's a bug we want to hear about).